pkg/apis/local/v1alpha1/localvolumeset_types.go

```diff
@@ -0,0 +1,118 @@
+package v1alpha1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	localv1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+// The mechanical properties of the devices
+const (
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+// DeviceType is the types that will be supported by the LSO.
+type DeviceType string
+
+const (
+	// RawDisk represents a device-type of block disk
+	RawDisk DeviceType = "RawDisk"
+	// Partition represents a device-type of partion
+	Partition DeviceType = "Partition"
+)
+
+// DeviceInclusionSpec holds the inclusion filter spec
+type DeviceInclusionSpec struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty no devices will be selected.
+	Devices []DeviceType `json:"devices"`
```

Comment:
DeviceTypes* looks like this name is a mistake that slipped through in the design too.

Reply:
No I think it should be `Devicetype` its a custom data type we created